projects/packages/search/src/class-rest-controller.php

```diff
@@ -213,7 +235,60 @@ public function get_search_results( $request ) {
 			sprintf( '/sites/%d/search', absint( $blog_id ) )
 		);
 		$response = Client::wpcom_json_api_request_as_user( $path, '1.3', array(), null, 'rest' );
-		return $this->make_proper_response( $response );
+		return rest_ensure_response( $this->make_proper_response( $response ) );
+	}
+
+	/**
+	 * Activate plan: activate the search module, instant search and do initial configuration.
+	 * Typically called from WPCOM.
+	 *
+	 * POST `jetpack/v4/search/plan/activate`
+	 *
+	 * @param WP_REST_Request $request - REST request.
+	 */
```

Comment:
Could we change this route to accept an empty payload instead of a `plan_info` payload? Given that this route accepts payloads from any admin user, I could see this route failing with malformed POST payloads.

Upon receiving the activation call, could we make a request to WPCOM for the search plan information (via the Plan class, I assume)?

Reply:
> Could we change this route to accept an empty payload instead of a plan_info payload? 

I should have written that in the PR description - it had been implemented like that initially and was changed later to accept a `plan_info` payload to optimize latency here. And plan info is constantly updated on Jetpack heartbeat too. So I'm not too worried about malformed payload, as the endpoint requires admin access. Even the module is turned on by some reason, the search API is still not available if there's no search subscription. 

I'll add a fallback, if the payload is not valid, we'd make a request to WPCOM.

Reply:
Addressed in b9a295d23d

Reply:
Great idea; looks like we're now optimized for both latency and resiliency 🙂